src/util/ChartUtils.ts

```diff
@@ -165,10 +166,18 @@ export const getMainColorOfGraphicItem = (item: ReactElement) => {
   return result;
 };
 
-interface FormattedGraphicalItem {
-  props: any;
+export interface FormattedGraphicalItem {
+  props: {
+    sectors?: ReadonlyArray<any>;
+    data?: ReadonlyArray<any>;
+  };
   childIndex: number;
-  item: any;
+  item: ReactElement<{ legendType?: LegendType; hide: boolean; name?: string; dataKey: unknown }>;
+}
+
+interface SectorOrDataEntry {
+  name: any;
+  fill: any;
```

Comment:
Could we not use any? (But also, the whole PR is already a great improvement, so this is fine, just wondering)

Reply:
Yep definitely - I just wanted to keep things limited in one PR.